spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/ZipkinAutoConfiguration.java

```diff
@@ -123,7 +126,7 @@ public EndpointLocator zipkinEndpointLocator() {
 
 		private DiscoveryClientEndpointLocator discoveryClientEndpointLocator() {
 			if (this.client!=null) {
-				return new DiscoveryClientEndpointLocator(this.client);
+				return new DiscoveryClientEndpointLocator(this.client, new EndpointCacheImpl(this.zipkinProperties));
```

Comment:
`EndpointCacheImpl` should be a bean that can have other implementations, otherwise, drop the interface.